src/test/groovy/com/gooddata/executeafm/afm/SimpleMeasureDefinitionTest.groovy

```diff
@@ -108,11 +113,78 @@ class SimpleMeasureDefinitionTest extends Specification {
         copy.getObjQualifier().getUri() == 'uri'
     }
 
+    def "should copy with uri converter"() {
+        when:
+        def definition = new SimpleMeasureDefinition(new IdentifierObjQualifier("id"))
+        def qualifiersConversionMap = [(new IdentifierObjQualifier("id")): new UriObjQualifier("uri")]
+        def copy = definition.withObjUriQualifiers({ identifierQualifier ->
+            return Optional.ofNullable(qualifiersConversionMap.get(identifierQualifier))
+        })
+
+        then:
+        copy.getUri() == 'uri'
+        copy.getItem().getUri() == 'uri'
+    }
+
+    def "should return the same object when copying with uri converter and uri is already used"() {
+        when:
+        def definition = new SimpleMeasureDefinition(new UriObjQualifier('uri'))
+        def copy = definition.withObjUriQualifiers({ identifierQualifier -> Optional.empty() })
+
+        then:
+        definition == copy
+    }
+
+    def "should fail when qualifier converter is not provided"() {
```

Comment:
merge with the test below using `where`

Reply:
Fixed. For other classes that had this issue as well.